File: Sliding Window/2958. Length of Longest Subarray With at Most K Frequency/2958.py
class Solution:
    def maxSubarrayLength(self, nums: List[int], k: int) -> int:
        freq = defaultdict(int)

        l=res=0

        for r in range(len(nums)):
            freq[nums[r]]+=1
            while freq[nums[r]]>k:
                freq[nums[l]]-=1
                l+=1
            res=max(res,r-l+1)
        return res
        

File: Sliding Window/2958. Length of Longest Subarray With at Most K Frequency/2958.cpp
class Solution {
public:
    int maxSubarrayLength(vector<int>& nums, int k) {
        ios_base::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);

        unordered_map<int,int> freq;
        int l=0;
        int res=0;

        for(int r=0; r<nums.size(); ++r) {
            freq[nums[r]]++;
            while(freq[nums[r]]>k) {
                freq[nums[l]]--;
                l++;
            }
            res=max(res,r-l+1);
        }
        return res;
    }
};